ando/tools/generator/nwb2bidsgenerator.py

```diff
@@ -0,0 +1,214 @@
+        generic_ephys_name = f'{subject_label}_{session_label}_'
+        sub_ses_dict[subject_label].append(session_label)
+        ses_path = subject_path/session_label
+        data_path = ses_path/'ephys'
+        data_path.mkdir(parents=True, exist_ok=True)
+
+        # move nwbfile
+        bep_nwbfile_path = data_path/(generic_ephys_name + 'ephys.nwb')
+        if move_nwb:
+            if not bep_nwbfile_path.exists():
+                nwb_file.replace(bep_nwbfile_path)
+        else:
+            if not bep_nwbfile_path.exists():
+                bep_nwbfile_path.symlink_to(nwb_file)
+
+        # channels.tsv:
+        bep_channels_path = data_path/(generic_ephys_name + 'channels.tsv')
+        if not bep_channels_path.exists():
+            channels_df.dropna(axis='columns', how='all', inplace=True)
+            channels_df.to_csv(bep_channels_path, sep='\t', index=False)
+
+        # probes/contacts.tsv:
+        bep_probes_path = data_path/(generic_ephys_name + 'probes.tsv')
+        if not bep_probes_path.exists():
+            probes_df.dropna(axis='columns', how='all', inplace=True)
+            probes_df.to_csv(bep_probes_path, sep='\t', index=False)
+        bep_contacts_path = data_path/(generic_ephys_name + 'contacts.tsv')
+        if not bep_contacts_path.exists():
+            contacts_df.dropna(axis='columns', how='all', inplace=True)
+            if len(contacts_df) > 0:
+                contacts_df.to_csv(bep_contacts_path, sep='\t', index=False)
+
+        # ephys.json:
+        bep_ephysdesc_path = data_path/(generic_ephys_name + 'ephys.json')
+        with open(bep_ephysdesc_path, 'w') as j:
+            json.dump(ephys_desc_json, j)
+
+        # create sessions.tsv
+        sessions_df.to_csv(bep_sessions_path, sep='\t', index=False)
+
+    # clean sessions_df_csv files for all subjects:
+    for sub_label in sub_ses_dict:
+        subject_path = output_path/sub_label/f'{sub_label}_sessions.tsv'
+        session_df_loop = pd.read_csv(subject_path, sep='\t')
+        session_df_loop.dropna(axis='columns', how='all', inplace=True)
+        session_df_loop.to_csv(subject_path, sep='\t', index=False)
+
+    # create participants.tsv:
+    participants_df.dropna(axis='columns', how='all', inplace=True)
+    participants_df.to_csv(
+        output_path/
+        'participants.tsv',
+        sep='\t',
+        index=False)
+
+    # create dataset_desrciption.json
+    with open(output_path/'dataset_description.json', 'w') as j:
+        if all([True for au in dataset_desc_json['Authors'] if au is None]):
+            _ = dataset_desc_json.pop('Authors')
+        dataset_desc_tosave = {k: v for k,
+                                        v in dataset_desc_json.items() if v is not None}
+        json.dump(dataset_desc_tosave, j)
+    print(
+        f'total nwbfiles orgainzed {file_count}, sessions count {sessions_count}')
+    # validate:
+    if validate:
+        is_valid(output_path)
+    return sub_ses_dict
```

Comment:
I am not sure if it makes more sense to return the output path here instead of the internal `sub_ses_dict`. This might be more useful when incorporating this script in a pipeline. Any opinion on this @Slowblitz @SylvainTakerkart ?

Reply:
I think it depends on what we want to do after in the pipeline

Reply:
it returns the path where the BIDS formatted dataset is located.